Backport of https://github.com/NixOS/nixpkgs/pull/577208 (brotli 1.2.0 -> 1.2.1,
merged to staging). 1.2.1 includes the CVE-2026-111497 fix, so nodejs's
brotli override that backported it is dropped (it would no longer apply).

diff --git a/pkgs/by-name/br/brotli/package.nix b/pkgs/by-name/br/brotli/package.nix
--- a/pkgs/by-name/br/brotli/package.nix
+++ b/pkgs/by-name/br/brotli/package.nix
@@ -12,13 +12,13 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "brotli";
-  version = "1.2.0";
+  version = "1.2.1";
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "brotli";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-kl8ZHt71v17QR2bDP+ad/5uixf+GStEPLQ5ooFoC5i8=";
+    hash = "sha256-llY72cYffDv0ugsH0p89xIPEbElV4L8u7U/Fe8hT3gI=";
   };
 
   patches = [
diff --git a/pkgs/development/web/nodejs/nodejs.nix b/pkgs/development/web/nodejs/nodejs.nix
--- a/pkgs/development/web/nodejs/nodejs.nix
+++ b/pkgs/development/web/nodejs/nodejs.nix
@@ -134,22 +134,8 @@
   useSharedZstd = lib.versionAtLeast version "22.15";
 
   sharedLibDeps = {
-    # TODO: remove this on staging and land with https://github.com/NixOS/nixpkgs/pull/576792
-    brotli =
-      if majorVersion == "24" then
-        brotli
-      else
-        brotli.overrideAttrs (old: {
-          patches = old.patches ++ [
-            # Fix for CVE-2026-111497
-            (fetchurl {
-              url = "https://github.com/google/brotli/commit/0ed4268a4401abcb4472096340477cf4af7d1a2b.patch?full_index=1";
-              hash = "sha256-1vliUJAAIj3vcGKVgim9CEV+XgMuftDR2U+3kLBUsfw=";
-            })
-          ];
-        });
-
     inherit
+      brotli
       libuv
       nghttp2
       nghttp3
